packages/esm-patient-chart-app/src/encounter-list/components/encounter-list.component.tsx

```diff
@@ -0,0 +1,296 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import { navigate, showModal, showSnackbar, Visit } from '@openmrs/esm-framework';
+import { EmptyState } from '@openmrs/esm-patient-common-lib';
+import { useTranslation } from 'react-i18next';
+import { EncounterListDataTable } from './table.component';
+import { Button, Link, OverflowMenu, OverflowMenuItem, DataTableSkeleton, Pagination } from '@carbon/react';
+import { Add } from '@carbon/react/icons';
+import { launchEncounterForm } from '../utils/helpers';
+import { deleteEncounter } from '../encounter-list.resource';
+import { useEncounterRows, useFormsJson } from '../hooks';
+
+import styles from './encounter-list.scss';
+import { type TableRow, type Encounter, type Mode } from '../types';
+
+export interface EncounterListColumn {
+  key: string;
+  header: string;
+  getValue: (encounter: Encounter) => string;
+  link?: any;
+}
+
+export interface EncounterListProps {
+  patientUuid: string;
+  encounterType: string;
+  columns: Array<any>;
+  headerTitle: string;
+  description: string;
+  formList?: Array<{
+    name?: string;
+    uuid: string;
+    excludedIntents?: Array<string>;
+    fixedIntent?: string;
+    isDefault?: boolean;
+  }>;
+  launchOptions: {
+    hideFormLauncher?: boolean;
+    displayText?: string;
+    workspaceWindowSize?: 'minimized' | 'maximized';
+  };
+  filter?: (encounter: Encounter) => boolean;
+  afterFormSaveAction?: () => void;
+  deathStatus?: boolean;
+  currentVisit: Visit;
+}
+
+export const EncounterList: React.FC<EncounterListProps> = ({
+  patientUuid,
+  encounterType,
+  columns,
+  headerTitle,
+  description,
+  formList,
+  filter,
+  launchOptions,
+  afterFormSaveAction,
+  currentVisit,
+  deathStatus,
+}) => {
+  const { t } = useTranslation();
+
+  const [currentPage, setCurrentPage] = useState(1);
+  const [pageSize, setPageSize] = useState(10);
+
+  const { formsJson, isLoading: isLoadingFormsJson, error: errorFormJson } = useFormsJson(formList[0].uuid);
```

Comment:
Wouldn't doing `formList[0].uuid` break in the (maybe unlikely) scenario that formList is an empty array?

Reply:
Also `errorFormJson` is unused so we could remove it
```suggestion
  const { formsJson, isLoading: isLoadingFormsJson } = useFormsJson(formList[0].uuid);
```